### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by pdx-cs-lbryan**
Merged with add/remove commits to master

### CODE REVIEW

File: form.py [hunk 1]
```diff
@@ -267,27 +275,27 @@ def review():
...
-                    db.session.commit()
-                except:
-                    displayError('submission error')
+                gbmodel.db_session.add(report)
+
+        try:
+            #students = gbmodel.students()
+            #sdt = students.query.filter_by(id=getID()).first()
+            sdt = gbmodel.db_session.query(gbmodel.students).filter_by(id=getID()).first()
+            state = sdt.active
```
Comment (J8cob1) [inline form.py]: It seems like the sdt.active check should take place after checking if std is null

File: gbmodel/model_sqlalchemy.py [hunk 2]
```diff
@@ -13,7 +14,7 @@ def getMaxTeamID(self):
             return 1
         else:
             return max_id[0] + 1
-
+
```
Comment (J8cob1) [inline gbmodel/model_sqlalchemy.py:17]: Extra spacing

File: app.py [hunk 3]
```diff
@@ -13,32 +13,23 @@
...
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = True
 db = SQLAlchemy(app)
 db.Model.metadata.reflect(db.engine)
 db_session = scoped_session(sessionmaker(bind=engine))
 
 app.add_url_rule('/',
                  view_func=Index.as_view('index'))
 
+#form blueprint, can change to url_rules without blueprint if desired
+app.register_blueprint(form_bp)
```
Comment (J8cob1) [inline app.py:31]: At some point, we might consider standardizing our strategy here (do we want to go all blueprints, all routes, etc) if we can. We can worry about this later though

File: form.py [hunk 4]
```diff
@@ -131,7 +140,8 @@ def review():
         cid = getCap()
 
         # get team members
-        mems = students.query.join(team_members).filter_by(tid=tid).distinct()
+        students = gbmodel.students()
+        mems = gbmodel.db_session.query(gbmodel.students).join(gbmodel.team_members).filter_by(tid=tid).distinct()
```
Comment (J8cob1) [inline form.py]: So not try-catch here? If this is intentional, that's fine

File: gbmodel/model_sqlalchemy.py (hunk 2 above)
Comment (J8cob1) [inline gbmodel/model_sqlalchemy.py:17]: Extra spacing. Not a big deal

File: form.py (hunk 4 above)
Comment (J8cob1) [inline form.py]: This call doesn't seem guarded by a try-catch statement. If it isn't, is one needed here?

File: form.py (hunk 1 above)
Comment (J8cob1) [inline form.py]: Shouldn't the std.active assignment come after the sdt is None check?

**Review by J8cob1 — CHANGES_REQUESTED** [review]
So

File: form.py (hunk 1 above)
Comment (pdx-cs-lbryan) [reply form.py]: addressed

File: gbmodel/model_sqlalchemy.py (hunk 2 above)
Comment (pdx-cs-lbryan) [reply gbmodel/model_sqlalchemy.py:17]: Not sure about this -- looks like it matches the others?

File: app.py (hunk 3 above)
Comment (pdx-cs-lbryan) [reply app.py:31]: yeah, I can change it if we want it all the other way, but that can come later

File: form.py (hunk 4 above)
Comment (pdx-cs-lbryan) [reply form.py]: fixed

File: gbmodel/model_sqlalchemy.py (hunk 2 above)
Comment (pdx-cs-lbryan) [reply gbmodel/model_sqlalchemy.py:17]: looks like the others?

File: form.py (hunk 4 above)
Comment (pdx-cs-lbryan) [reply form.py]: fixed

File: form.py (hunk 1 above)
Comment (pdx-cs-lbryan) [reply form.py]: fixed

File: gbmodel/model_sqlalchemy.py (hunk 2 above)
Comment (J8cob1) [reply gbmodel/model_sqlalchemy.py:17]: Not important either way

File: form.py [hunk 5]
```diff
@@ -267,27 +278,30 @@ def review():
...
-                        sdt.final_done = 1
-                    else:
-                        displayError('submitting reports not open')
-
-                    db.session.commit()
-                except:
-                    displayError('submission error')
+                gbmodel.db_session.add(report)
+
+        try:
```
Comment (J8cob1) [inline form.py:283]: At some point, I might want to know how this works, but my lack of it won't stop me from approving

**Review by J8cob1 — APPROVED** [review]
Approving

**Pull request merged**